coffea/jetmet_tools/CorrectedJetsFactory.py

```diff
@@ -98,8 +98,21 @@ def getfunction(layout, depth):
     return smearfact
 
 
+def get_corr_inputs(jets, corr_obj, name_map):
+    """
+    Helper function for getting values of input variables
+    given a dictionary and a correction object.
+    Source:
+    https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/blob/master/examples/jercExample.py?ref_type=heads
+    """
+    input_values = [
+        awkward.flatten(jets[name_map[inp.name]]) for inp in corr_obj.inputs
+    ]
+    return input_values
+
+
 class CorrectedJetsFactory(object):
-    def __init__(self, name_map, jec_stack):
+    def __init__(self, name_map, jec_stack, jec_names, jec_year):
```

Comment:
Please do not alter the signature of this function (especially with new non-defaulted arguments). We want to maintain backwards compatibility for the time being. 

Could you try implementing this such that the code detects correction-lib inputs coming in from the JEC stack, and then follow a different code path in the case of correctionlib inputs?

Reply:
@lgray thank you for pointing out this. Would it work if jec_stack is a list or a dict if one wants to use `correction-lib`? IMU, the JECStack class (as well as the auxiliary ones) is not needed when using `correction-lib`, as the latter already functionalities to get the right numbers from the formulas that are part of the JME corrections/

Reply:
The JEC stack also defines the mapping from names in your data to the arguments of the functions to evaluate. Sure, you can have the `jec_stack` be something much simpler (like a list) if correctionlib does a bunch of the bookkeeping for you.

Though I think you'll need to do some arranging yourself for the JES uncertainties.